internal/exec/stages/disks/partitions.go

```diff
@@ -353,7 +354,23 @@ func (s stage) partitionDisk(dev types.Disk, devAlias string) error {
 		case exists && shouldExist && matches:
 			s.Logger.Info("partition %d found with correct specifications", part.Number)
 		case exists && shouldExist && !wipeEntry && !matches:
-			return fmt.Errorf("Partition %d didn't match: %v", part.Number, matchErr)
+			number := part.Number == info.Number
+			label := part.Label == nil || *part.Label == info.Label
+			start := part.StartSector == nil || *part.StartSector == info.StartSector
+			guid := part.GUID == nil || *part.GUID == "" || strings.ToLower(*part.GUID) == strings.ToLower(info.GUID)
+			typeGUID := part.TypeGUID == nil || *part.TypeGUID == "" || strings.ToLower(*part.TypeGUID) == strings.ToLower(info.TypeGUID)
+			if resize && number && label && start && guid && typeGUID {
+				s.Logger.Info("partition %d found with resize option, resizing partition", part.Number)
```

Comment:
It's good to try to write log messages from a user's perspective, rather than the code's perspective.  `resizing partition %d` should be sufficient.